llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4733,6 +4755,36 @@ LoopVectorizationCostModel::computeMaxVF(ElementCount UserVF, unsigned UserIC) {
   // FIXME: look for a smaller MaxVF that does divide TC rather than masking.
   if (Legal->prepareToFoldTailByMasking()) {
     CanFoldTailByMasking = true;
+    if (getTailFoldingStyle() == TailFoldingStyle::None)
+      return MaxFactors;
+
+    if (UserIC > 1) {
+      LLVM_DEBUG(dbgs() << "LV: Preference for VP intrinsics indicated. Will "
+                           "not generate VP intrinsics since interleave count "
+                           "specified is greater than 1.\n");
+      return MaxFactors;
+    }
+
+    // FIXME: use actual opcode/data type for analysis here.
+    PreferEVL = MaxFactors.ScalableVF.isScalable() &&
```

Comment:
Would be good to leave a comment here explaining why this is limited to scalable VFs only for now. After briefly looking at the PPC implementation for `hasActiveVectorLength` we would need to check if all loads & stores are of supported types

Might also be good to have a test for PPC to cover this path

Reply:
1. Added FIXME.
2. Added tests for PPC.